src/main/java/eu/donals/UserInterface.java

```diff
@@ -0,0 +1,261 @@
+package eu.donals;
+
+import com.googlecode.lanterna.Symbols;
+import com.googlecode.lanterna.TerminalPosition;
+import com.googlecode.lanterna.TerminalSize;
+import com.googlecode.lanterna.TextCharacter;
+import com.googlecode.lanterna.graphics.TextGraphics;
+import com.googlecode.lanterna.input.KeyStroke;
+import com.googlecode.lanterna.input.KeyType;
+import com.googlecode.lanterna.screen.Screen;
+import com.googlecode.lanterna.screen.TerminalScreen;
+import com.googlecode.lanterna.terminal.DefaultTerminalFactory;
+import com.googlecode.lanterna.terminal.Terminal;
+import com.googlecode.lanterna.terminal.swing.SwingTerminal;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserInterface {
+    private Terminal terminal;
+    private Screen screen;
+    private TextGraphics tg;
+
+    private static UserInterface instance;
+
+    static {
+        try {
+            instance = new UserInterface();
+        } catch (IOException e) {
+            e.printStackTrace();
+            System.exit(1);
+        }
+    }
+
+    private UserInterface() throws IOException {
+        terminal = new DefaultTerminalFactory().setPreferTerminalEmulator(true).setInitialTerminalSize(new TerminalSize(160, 40)).createTerminal();
+        screen = new TerminalScreen(terminal);
+        tg = screen.newTextGraphics();
+        screen.setCursorPosition(null);
+        screen.startScreen();
+    }
+
+    public static UserInterface getInstance() { return instance; }
+
+    // displays the game initialisation, returns the character's name
+    public String initGame() throws IOException, InterruptedException {
+        String presentationText  = "Team 15 studios present...";
+        String gameTitle = "Super Ultimate Pizza Maker 69";
+        String playernamePrompt = "Please enter your character's name: ";
+        String userInput = "";
+        int cursorX = (centreText(160, playernamePrompt) + playernamePrompt.length() + 1);
+        int cursorY = 10;
+        TerminalPosition initialPromptPosition = new TerminalPosition(cursorX, cursorY);
+
+        tg.putString(centreText(160, presentationText), 10, presentationText);
```

Comment:
Some pieces of a few lines of code like this one and the one below are repeated multiple times. They could be separate functions to make this function a bit smaller.